internal/providers/terraform/dir_provider.go

```diff
@@ -470,39 +474,48 @@ func checkTerraformVersion(v string, fullV string) error {
-		msg += "\nTerraform requires AWS credentials to be set.\n"
-		msg += "For more details, see https://registry.terraform.io/providers/hashicorp/aws\n"
+
+	if strings.HasPrefix(stderr, "Error: Failed to select workspace") {
+		msg += "\n\nYou have two options:\n"
+		msg += "1. Run `terraform workspace select your_workspace` first or set the TF_WORKSPACE environment variable.\n\n"
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if errors.Is(err, ErrMissingCloudToken) || errors.Is(err, ErrInvalidCloudToken) || strings.HasPrefix(stderr, "Error: Required token could not be found") {
+		msg += "\n\nYou have two options:\n"
+		msg += "1. Run `terraform login` or set the INFRACOST_TERRAFORM_CLOUD_TOKEN environment variable.'\n\n"
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if strings.HasPrefix(stderr, "Error: No value for required variable") {
+		msg += "\n\nYou have two options:\n"
+		msg += "1. Pass the variables using the --terraform-plan-flags option,\n     e.g. --terraform-plan-flags=\"-var-file=my.tfvars\"\n\n"
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if strings.HasPrefix(stderr, "Error: Failed to read variables file") {
+		msg += "\n\nYou have two options:\n"
+		msg += "1. Ensure the variable file is passed relative to your Terraform directory,\n     e.g. --path=path/to/code --terraform-plan-flags=\"-var-file=my.tfvars\"\n\n"
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if strings.HasPrefix(stderr, "Error: error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.") {
+		msg += "\n\nTerraform requires AWS credentials to be set. You have two options:\n"
+		msg += fmt.Sprintf("1. See %s for details of how to set credentials.\n\n", ui.LinkString("https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication"))
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if p.IsTerragrunt {
+		msg += fmt.Sprintf("\n\nSee %s for how to generate multiple Terraform plan JSON files for your Terragrunt project, and use them with Infracost.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else {
+		msg += fmt.Sprintf("\n\nTry setting the --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+
 	}
 
-	fmt.Fprintln(os.Stderr, msg)
+	return fmt.Errorf("%v%s", err, msg)
```

Comment:
Since we're not printing these to directly os.Stderr anymore it's more possible to add tests for them. I've added a couple of examples, but added every case as part of this PR.